src/controllers/requests/editRequests.js

```diff
@@ -0,0 +1,44 @@
+const db = require('../../database/db');
+require('dotenv').config();
+
+const getRequestsById = async (id) => {
+    const query = 'SELECT * from requests WHERE id = $1';
+    const response = await db.query(query, [id]);
+    return response.rows[0];
+};
+
+const editRequest = async (request, response) => {
+    try{
+        const {details, id} = request.body;
+        const existingRequest = await getRequestsById(id);
+        if (!existingRequest){
+            return response.status(401).json({
+                status: "error",
+                message: "Request does not exist"
+            })
+        }
+
+        if (existingRequest.userid !== request.user.id) {
+            return response.status(403).json({
+                status: "error",
+                message: "Permission denied: You can only edit your own requests"
+            });
+        }
+
+        const updateQuery = 'UPDATE requests SET details = $1 WHERE id = $2';
+        await db.query(updateQuery, [details, id]);
+
+
+        return response .status(201).json({
```

Comment:
The status code should be 200. 201 is for when something new is being created, but in this scenario, we're not creating something new, we're updating something that already exists